networkx/algorithms/centrality/group.py

```diff
@@ -515,7 +513,10 @@ def _heuristic(k, root, DF_tree, D, nodes, greedy):
             zip(np.diag(DF_tree.nodes[node_p]["betweenness"]), nodes), reverse=True
         )
     ]
-    [CL.remove(m) for m in CL if m in DF_tree.nodes[node_p]["GM"]]
+    added_nodes = DF_tree.nodes[node_p]["GM"]
+    for m in CL:
+        if m in added_nodes:
+            CL.remove(m)
```

Comment:
It looks like this would be clearer and probably more efficient to include in the previous list comprehension which defines the `CL` list we are removing from here.  If I understand correctly, it would just need an additional line after 514 with something like:
```python
if node not in DF_tree.nodes[node_p]["GM"]
```